scripts/install_linux.sh

```diff
@@ -0,0 +1,126 @@
+#!/bin/bash -e
+
+get_machine_arch () {
+    machine_arch=""
+    case $(uname -m) in
+        i386)     machine_arch="386" ;;
+        i686)     machine_arch="386" ;;
+        x86_64)   machine_arch="amd64" ;;
+        arm64)    machine_arch="arm64" ;;
+        aarch64)  dpkg --print-architecture | grep -q "arm64" && machine_arch="arm64" || machine_arch="arm" ;;
+    esac
+    echo $machine_arch
+}
+arch=$(get_machine_arch)
+
+echo "arch=$arch"
+
+remote_filename="tfsec"
+local_filename="tfsec"
+case "$(uname -s)" in
+  Darwin*)
+    remote_filename+="-darwin-${arch}"
+    ;;
+  MINGW64*)
+    remote_filename+="-windows-${arch}"
+    local_filename+=".exe"
+    ;;
+  MSYS_NT*)
+    remote_filename+="-windows-${arch}"
+    local_filename+=".exe"
+    ;;
+  *)
+    remote_filename+="-linux-${arch}"
+    ;;
+esac
+checksum_file="tfsec_checksums.txt"
+download_path="/tmp/tfsec"
```

Comment:
Ok I did not get why the location of the checksum file should be hidden. Am I missing something obvious?

Reply:
Submitted https://github.com/aquasecurity/tfsec/pull/1870.

It's not about hiding the location. It's because `/tmp` can't be trusted to be secure and could result in a TOCTOU race condition issue. You're downloading a file and then escalating privileges to install it as root. An unprivileged user on the machine could use a race condition to replace the files in `/tmp` so that they are installed instead of the actual tfsec install.